active/0000-plaforms.md

```diff
@@ -0,0 +1,328 @@
+  "ember-addon",
+
+  // if present, indicates this addon contains process hooks for ember-cli-platforms
+  "ember-cli-platform-plugin",
+  
+  // if present, indicates this addon contains a platform implementation for ember-cli-platforms
+  "ember-cli-platform",
+  
+  // if present, indicates this addon implements deployment hooks
+  "ember-cli-platform-deploy"
+]
+```
+
+
+#### Implement one or more pipeline hooks
+
+In order for a plugin to be useful it must implement one or more of the ember-cli-platforms
+pipeline hooks.
+
+To do this you must implement a function called `createPlatformPlugin` in your `index.js` file.
+This function must return an object that contains:
+
+1. A name property which is what your plugin will be referred to.
+2. One or more implemented pipeline hooks
+
+**Example:**
+
+```js
+module.exports = {
+  name: 'ember-cli-platform-live-reload',
+
+  createPlatformPlugin: function(options) {
+    return {
+      name: "live-reload",
+
+      didBuild: function(context) {
+        //do something amazing here once the project has been built
+      }
+    };
+  }
+
+};
+```
+
+#### Process Hooks
+
+- configure
+- setup
+- willBuild
+- build
+- didBuild
+- teardown
+
+
+
+### Creating a Deployment Plugin
+
+This is built overtop of [ember-cli-deploy](http://ember-cli.com/ember-cli-deploy/docs/v0.5.x/pipeline-hooks/),
+but integrates with your platform flow.
+
+To create a deploy plugin, follow the same steps as for `process hooks`, making sure to
+add the keyword `ember-cli-platform-deploy` to `package.json`. The following is a full list
+of hooks which can be returned by `createPlatformPlugin` for use with deployment.
+
+#### All Hooks (including deployment)
+
+- configure
+- setup
+- willDeploy
+- willBuild, build, didBuild,
+- willPrepare, prepare, didPrepare,
+- willUpload, upload, didUpload,
+- willActivate, activate, didActivate, (only if --activate flag is passed)
+- didDeploy,
+- teardown
+
+
+
+### Creating a Platform Plugin
+
+Platform plugins have several concerns they need to manage.  They should
+
+- implement the build hooks of the pipeline
+- provide a starter kit for the platform
+- provide a CLI proxy for SDK commands
+- provide a blueprint for project installation
+- run any necessary SDK installation commands
+- handle SDK specific file inclusion
+
+**Command Runners**
+
+- ember-cli-platform supplied a commandRunner you can use to run CLI commands directly from
+  the pipeline if necessary.
+
+**Command Proxies**
+
+- If your ember-cli-platform-plugin specifies a `commandName`, it will be used to generate
+ a command proxy for that platform.  For instance:
+ 
+```cli
+{
+  commandName: 'cordova'
+}
+```
+
+If a user generates a cordova platform with the following command
+
+```cli
+ember platform cordova -t "ios"
+```
+
+Then an `npm` command script will be generated and installed that proxies use of
+`cordova-ios` into the cordova project directory ios, allowing you to easily use
+the cordova-cli.
```

Comment:
Projects such as cordova come with their own cli and own commands to run, ember-cli-cordova attempted to proxy these commands through an `ember` command, but that results in a very large overhead to setting up and running the command.

This would be a simple script which allows you to run that platform's commands from your ember project directory instead of needing to change directories.  It's a convenience factor which probably should be taken out of scope and left to an addon.